src/odemis/driver/tescan.py

```diff
@@ -1096,6 +1113,57 @@ def _pollXYZ(self):
         except Exception:
             logging.exception("Unexpected failure during XYZ polling")
 
+    def _checkPosition(self, ref_pos: Dict[str, float], pos: Dict[str, float], rel: bool = False):
+        """
+        Checks if the current stage position is equal to the given position.
+        :param ref_pos: stage position before movement
+        :param pos: requested stage position in relative or absolute coordinates
+        :param rel: By default False, requested stage position is in absolution position otherwise relative
+        """
+        # Get the target position in absolute coordinates
+        if rel:
+            target_pos = {ax: val + pos[ax] for ax, val in pos.items()}
+        else:
+            target_pos = {ax: pos[ax] for ax, val in pos.items()}
+
+        # Drop axes from the original position, which are not important because they have not moved
+        orig_pos = {a: ref_pos[a] for a, nv in target_pos.items() if nv != ref_pos[a]}
+
+        # TODO: base the timeout on stage movement time estimation, needs to be checked on hardware
+        wait_stage_move = 5  # s
+        expected_end_time = time.time() + wait_stage_move  # s
+
+        # Update rotational and linear tolerances according to the magnitude of requested change
+        linear_axes_to_check = {"x", "y", "z"}.intersection(target_pos.keys())
+        rotational_axes_to_check = {"rx", "rz"}.intersection(target_pos.keys())
+        current_pos = self._position
+        movement_req = {ax: STAGE_FRACTION_TOTAL_MOVE * abs(target_pos[ax] - current_pos[ax]) for ax in
+                        target_pos.keys()}
+        tol_linear = STAGE_TOL_LINEAR
+        tol_rotation = STAGE_TOL_ROTATION
+        if linear_axes_to_check:
+            movement_req_linear = [movement_req[ax] for ax in linear_axes_to_check]
+            tol_linear = max(min(movement_req_linear), STAGE_TOL_LINEAR)  # m
+
+        if rotational_axes_to_check:
+            movement_req_rotational = [movement_req[ax] for ax in rotational_axes_to_check]
+            tol_rotation = max(min(movement_req_rotational), STAGE_TOL_ROTATION)  # radians
+
+        axes_to_check = linear_axes_to_check | rotational_axes_to_check
+
+        while not isNearPosition(current_pos=current_pos, target_position=target_pos,
```

Comment:
If you call this function at the end of the move, it doesn't make sense to keep waiting. Especially, it doesn't seem that the ._position has any chance to update...